pkg/kubelet/cm/dra/plugin/registration.go

```diff
@@ -179,8 +179,9 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 
 	// Storing endpoint of newly registered DRA Plugin into the map, where plugin name will be the key
 	// all other DRA components will be able to get the actual socket of DRA plugins by its name.
-	if draPlugins.add(pluginInstance) {
-		logger.V(1).Info("Already registered, previous plugin was replaced")
+
+	if oldPlugin, replaced := draPlugins.add(pluginInstance); replaced {
+		logger.V(1).Info("Already registered, plugin was replaced and will be forgotten by the kubelet till the next kubelet restart", "oldEndpoint", oldPlugin.endpoint)
```

Comment:
```suggestion
		logger.V(1).Info("DRA plugin already registered, the old plugin was replaced and will be forgotten by the kubelet till the next kubelet restart", "oldEndpoint", oldPlugin.endpoint)
```

"[Plugin] already registered, plugin was replace..." made me wonder *which* plugin got replaced.

Reply:
applied

Reply:
Agreed.

BTW, @klueska had observed that after shutting down a DRA driver, kubelet did not unregister the plugin. No issue filed yet, I haven't tried to reproduce myself. I'm mentioning it because you may have some context and it is related.

Reply:
If the DRa driver is shutdown cleanly, the resources slices do get removed. If not shutdown cleanly, then the socket file sticks around "forever" and the slices never get deleted by the kubelet.

Reply:
=> https://github.com/kubernetes/kubernetes/issues/128696